base_decorator.rb

```diff
@@ -0,0 +1,12 @@
+require_relative './nameable'
+
+class Decorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
```

Comment:
- [ ] Kindly make sure `setter` and `getter` is used on the `@nameable` variable i.e. `attr_accessor :nameable`

Reply:
@tresorsawasawa I just implemented the changes you requested even though setter and getter are not part of the requirements. I am sorry I am going to include these forced changes requested by you in my review.